tests/raftstore/cluster.rs

```diff
@@ -580,22 +580,29 @@ impl<T: Simulator> Cluster<T> {
         panic!("request failed after retry for 20 times");
     }
 
-    pub fn get_region(&self, key: &[u8]) -> metapb::Region {
+    pub fn get_region_not_eq(&self, key: &[u8], region: Option<&metapb::Region>) -> metapb::Region {
```

Comment:
I think `wait_for_region_update` is a better name.

Reply:
maybe using a filter function is better here? like

```rust
get_region_with(*self, key: &[u8], f: Fn(region: &metapb::Region) -> bool) -> metapb::Region
```

if filter function return true, we can get the region.